dali/kernels/tensor_shape.h

```diff
@@ -495,6 +495,10 @@ struct TensorListShapeBase {
     return out;
   }
 
+  void set_tensor_shape(int64_t sample, const TensorShape<sample_ndim> &sample_shape) {
```

Comment:
Why is this necessary?

Reply:
This is to allow `tls.set_tensor_shape(i, {1, 2, 3});` without compiler complaining for ambiguity. And I am not keen on using initializer_list as it has weird rules of promotions.

Reply:
Fair enough.

Reply:
Done